api/envoy/extensions/filters/common/matcher/action/v3/skip_action.proto

```diff
@@ -14,6 +14,9 @@ option (udpa.annotations.file_status).package_version_status = ACTIVE;
 
 // [#protodoc-title: Common Match Actions]
 
-// Indicates that the associated filter should be skipped.
+// Configuration for the SkipFilter match action. When matching results in this action, the
+// associated filter will be ignored going for all data callbacks (including the one that
+// triggered the match result). As a result, if this matching is done on the first data
+// callback (e.g. request headers), this will completely skip the filter.
```

Comment:
This is the first time the concept of "data callback" comes up. Is there a way to explain it more clearly?

What happens when the matcher is used in a non-HTTP filter context? E.g. in some new router scheme. I guess HTTP filter callbacks still apply there, but what about TAP, etc.?

Reply:
In my mind this would only be used in the context of HTTP filters (hence the name `SkipFilter`). Right now we don't have a mechanism to allowlist which actions can be used in various contexts, but I imagine we'll want this or some kind of template trickery to create unique sets of actions for different contexts. 

I've removed the use of "data callback", hopefully this will be easier to understand now